components/service-operator/apis/access/v1beta1/principal_types.go

```diff
@@ -70,6 +70,8 @@ type PrincipalSpec struct {
 	Secret string `json:"secret,omitempty"`
 	// Name of a service account to trust with access to this Principal, if any.
 	TrustServiceAccount string `json:"trustServiceAccount,omitempty"`
+	// Boolean to allow exposing credentials into the Secret.
+	ExposeCredentials bool `json:"exposeCredentials,omitempty"`
```

Comment:
Hate to get out the bike-shed paint brushes ... but `ExposeCredentials` sounds like a pretty terrifying and un-descriptive name :laughing:  .... how about `GenerateAccessSecrets` ?

Reply:
It was slightly deliberately scarily named, but I may have overdone it. We can change it to that though I don't know if it's scary enough. What shade should this bikeshed be exactly?

Reply:
don't think it needs to be scarey, descriptive more important than scarey